From 475555234ef3aded7b667f05866f8485991cd5fd Mon Sep 17 00:00:00 2001
From: Heinrich Toews <heinrich.toews@wago.com>
Date: Mon, 24 Jul 2017 16:38:09 +0000
Subject: [PATCH] tty: serial_core: let the driver handle TIOC[GS]RS485 ioctls
 as used in older kernels

	The following commits of Ricardo Ribalda Delgado introduced ioctls handling
	on serial core layer, firstly in Linux 3.19:

		a5f276f10 serial_core: Handle TIOC[GS]RS485 ioctls
		dadd7ecbf serial/omap: Use the rs485 functions on serial_core

	As our well tested RTU driver (omap-serial-rtu.c) is based on legacy omap serial
	RS485 support from 3.6/3.18 and updates to it would eventually make the driver
	unstable we prever at this point to use the legacy driver version instead of serial
	core implementation.

	This commit disables TIOC[GS]RS485 ioctls on serial core layer.
---
 drivers/tty/serial/serial_core.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/tty/serial/serial_core.c b/drivers/tty/serial/serial_core.c
index 6cbd46c..7913f02 100644
--- a/drivers/tty/serial/serial_core.c
+++ b/drivers/tty/serial/serial_core.c
@@ -1368,14 +1368,6 @@ uart_ioctl(struct tty_struct *tty, unsigned int cmd, unsigned long arg)
 	case TIOCSERGETLSR: /* Get line status register */
 		ret = uart_get_lsr_info(tty, state, uarg);
 		break;
-
-	case TIOCGRS485:
-		ret = uart_get_rs485_config(uport, uarg);
-		break;
-
-	case TIOCSRS485:
-		ret = uart_set_rs485_config(uport, uarg);
-		break;
 	default:
 		if (uport->ops->ioctl)
 			ret = uport->ops->ioctl(uport, cmd, arg);
-- 
2.7.4

